src/App.js

```diff
@@ -1,41 +1,77 @@
-import Button from './components/Button';
+import Form from './components/Form';
 import MainWrapper from './components/MainWrapper';
 import RecipeList from './components/RecipeList';
 import { useState } from 'react';
 
 export default function App() {
   const [recipes, setRecipes] = useState([]);
+  const [missingIngredients, setMissingIngredients] = useState([]);
   const { REACT_APP_API_KEY } = process.env;
-  let randomNumber1 = Math.floor(Math.random() * 5);
-  let randomNumber2 = Math.floor(Math.random() * 5);
-  const ingredients1 = ['tuna', 'beef', 'eggs', 'milk', 'pear'];
-  const ingredients2 = ['apple', 'salmon', 'chicken', 'lettuce', 'tofu'];
-  const urlIngredients = `https://api.spoonacular.com/recipes/findByIngredients?apiKey=${REACT_APP_API_KEY}&ingredients=${ingredients1[randomNumber1]},+${ingredients2[randomNumber2]}&number=1`;
+  let urlIngredients = '';
 
-  function handleRandomize() {
+  function onSubmitIngredients(arr) {
     async function getNewRecipe() {
       try {
+        // Check which URL should be send depending und number of ingredients
```

Comment:
Just one little thing: to me it looks like you have a typo in your comment.

Reply:
ah yeah thanks i will correct that ;D